libfreerdp/codec/color.c

```diff
@@ -505,9 +505,23 @@ BOOL freerdp_image_copy(BYTE* pDstData, DWORD DstFormat,
 					        &srcLine[xSrcOffset], copyDstWidth);
 				}
 			}
-			/* Source and destination are equal... */
-			else
+			/* Source and destination are equal... - maybe a in-place flip is required */
+			else if (vSrcVFlip)
 			{
+				int half = nHeight / 2;
+				BYTE* topLine = pDstData;
+				BYTE* tmpBuf = (BYTE*)_aligned_malloc(nDstStep, 16);
+				BYTE* bottomLine = pDstData + (nDstStep * (nHeight - 1));
+
+				for (y = 0; y < half; ++y)
+				{
+					CopyMemory(tmpBuf, topLine, nDstStep);
+					CopyMemory(topLine, bottomLine, nDstStep);
+					CopyMemory(bottomLine, tmpBuf, nDstStep);
+					topLine += nDstStep;
+					bottomLine -= nDstStep;
+				}
+				_aligned_free(tmpBuf);
```

Comment:
@mfleisz If you add vSrcVFlip in this branch, it may be a good idea to add this for non overlapping copy too